src/app.js

```diff
@@ -3,11 +3,255 @@ import $ from 'jquery';
+};
+
+/////////////////////////////////////////////////////////////////
+//////////////////////// DISPLAY ERRORS /////////////////////////
+/////////////////////////////////////////////////////////////////
+
+const errorTemplate = _.template($('#error-template').html());
+
+// Now I need to display the template and then append it to the place that i want to append it to
+const displayErrors = (errors) => {
+  $('#display-errors').empty();
+
+  let errorObject = {};
+  for (let key in errors) {
+    errorObject[key] = errors[key];
+  }
+
+  let generatedHtml = errorTemplate(errorObject);
+  $('#display-errors').append(generatedHtml);
+  $('.close-button').on('click', (event) => {
+    $('#display-errors').empty();
+  });
+};
+
+/////////////////////////////////////////////////////////////////
+////////////////////////////// SORT /////////////////////////////
+/////////////////////////////////////////////////////////////////
+
+const SORT_FIELDS = ['id', 'name', 'continent', 'about', 'category', 'weeks', 'cost']
+
+// Attach sorting click handlers
+const sortingHandler = () => {
+  SORT_FIELDS.forEach((field) => {
+    const headerElement = $(`.sort.${ field }`);
+    console.log(field);
+    // Attaching an event handler here for the .on function for each of the header elements
+    headerElement.on('click', (event) => {
+      // Comparator is a property
+      tripList.comparator = field;
+      // If you don't change the comparator it will resort based on the sort not the comparator
+      tripList.sort();
+      // console.log('HITS SORT FUNCTION');
+    });
+  });
+};
+
+
+
+/////////////////////////////////////////////////////////////////
+//////////////////////////// FILTER /////////////////////////////
+/////////////////////////////////////////////////////////////////
+
+const filterDropdown = () => {
+  let value = $("#filter")[0].selectedIndex;
+  return value;
+}
+
+const filterResults = () => {
+  let input, filter, table, tr, td, i;
+  input = $('#search-bar')[0];
+  filter = input.value.toUpperCase();
+  table = $('#all-trips')[0];
+  tr = table.getElementsByTagName('tr');
+
+  // Run this function to get the current value of the selected index
+  let index = filterDropdown();
+  // console.log("Value of the index: " + index);
+  for (i = 0; i < tr.length; i++) {
+    td = tr[i].getElementsByTagName('td')[index];
+    if (td) {
+      if (td.innerHTML.toUpperCase().indexOf(filter) > -1) {
+        tr[i].style.display = "";
+      } else {
+        tr[i].style.display = "none";
+      }
+    }
+  };
+};
+
+/////////////////////////////////////////////////////////////////
+///////////////////////// DOCUMENT READY ////////////////////////
+/////////////////////////////////////////////////////////////////
+
+
+$(document).ready(() => {
+
+
+
+  $('#create-trip-form').hide();
+  tripList.fetch();
+
+  // Listen and register. Once an update has been heard, render all the collection into the template
+  tripList.on('update', renderList);
```

Comment:
Minor note:  It's theoretically possible (1 in a trillion chance) that the `fetch` could finish and return before this line being executed.  In which case `renderList` would not be executed.  It's more proper to establish your listeners and then run `fetch`.